pkg/operator/ceph/object/bucket/rgw-handlers.go

```diff
@@ -47,15 +50,25 @@ func (p *Provisioner) createCephUser(username string) (accKey string, secKey str
 	return u.Keys[0].AccessKey, u.Keys[0].SecretKey, nil
 }
 
+func random(fsid string) string {
+	randString := ""
+	for range fsid {
+		rand, _ := rand.Int(rand.Reader, big.NewInt(int64(len(fsid))))
+		randString += string(fsid[rand.Int64()])
+	}
+	return string(randString)
+}
+
 func (p *Provisioner) genUserName(obcName, obcNamespace string) (string, error) {
 	// user name can be deterministically generated by obc name and namespace. since there can't be
 	// 2 obcs in the same namespace w/ the same name, this won't collide w/in the same k8s cluster.
 	// However, it can collide in multisite setups, so use the Ceph cluster FSID
+	// However, it can also collide with external clusters, so use random
 	fsid := p.clusterInfo.FSID
 	if p.clusterInfo.FSID == "" {
 		return "", errors.Errorf("failed to find ceph cluster FSID")
 	}
-	return "obc-" + obcNamespace + "-" + obcName + "-" + fsid, nil
+	return "obc-" + obcNamespace + "-" + obcName + "-" + random(fsid), nil
```

Comment:
How about using the OBC's uid instead? This would also solve issue #11345. Note that above comment

> there can't be 2 obcs in the same namespace w/ the same name, this won't collide w/in the same k8s cluster.

isn't true. 2 OBCs absolutely can collide if the first one was deleted with `reclaimPolicy: Retain`.

Reply:
I created #12884 which implements my suggestion.